### PR TITLE
refactor: response headers part

### PR DESCRIPTION
**Pull request by dannypsnl**
- move content-type helper into helper file
- private the headers contains by response.Response
- create SetHeaders method for Response
    rewrite handler part to fit

### CODE REVIEW

File: response/response.go [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 
 func (res *Response) WithHeaders(headers Headers) *Response {
 	for k, v := range headers {
-		res.Headers[k] = v
+		res.headers[k] = v
 	}
 	return res
 }
+
+func (res *Response) SetHeaders(w http.ResponseWriter) {
```
Comment (shana0440) [inline response/response.go]: I'm worried user will confuse WithHeaders & SetHeaders.

Comment (dannypsnl) [reply response/response.go]: I think the problem is `rocket` & `response` are dependent on each others

Comment (dannypsnl) [reply response/response.go]: We can:
- move response back into rocket.
- create Handle function for Response type

Comment (shana0440) [reply response/response.go]: What create Handler function for Response type mean?

Comment (dannypsnl) [reply response/response.go]: `Handle(w http.ResponseWriter)`

This solution has a problem is we are going to depend on package rocket. Then get a cycle.

Comment (shana0440) [reply response/response.go]: I think Handle is better than SetHeaders.
And I don't see any cycle import problem in here.

Comment (dannypsnl) [reply response/response.go]: We need to assert return type html, json these stuffs

Comment (shana0440) [reply response/response.go]: oh, I see.

Comment (shana0440) [reply response/response.go]: If we extract type to a subpackage, it will be better?

Comment (dannypsnl) [reply response/response.go]: Then it became hard to use, that's what I want to avoid.

Comment (dannypsnl) [reply response/response.go]: If we move it into `response`?

Comment (shana0440) [reply response/response.go]: oh, that will be great.

**Pull request merged**